### PR TITLE
Fix react build, ditch browserify-shim

### PR DESCRIPTION
**Pull request by chmanie**
Hi! First of all thank you for putting all this together and sorry for opening a PR just like that. It should serve more like a starting point for a discussion.

I had problems in the past to [include the React component into my project](https://github.com/nosir/cleave.js/issues/111). So I just set the main entry point of the project to the transpiled `./dist/cleave.js`. Also I removed the necessity of transpiling from the react entry point `cleave.js/react`.

A side effect of this is that we can get rid of the browserify instructions from the package.json so people who don't use browserify-shim and/or babelify [don't have to install it anymore](https://github.com/nosir/cleave.js/issues/72#issuecomment-237596018).

I looked into the gulp-tasks/browserify.js and tried to figure out what you are using `browserify-shim` for but couldn't do it. Also even in master the `js:react:watch` task doesn't work for me. Is it even being used?

Thanks again, hoping for your feedback!

### CODE REVIEW

**Comment by nosir** [comment]
@chmanie Sorry, was in a long vacation, will have a look and get back to this soon.

File: package.json [hunk 1]
```diff
@@ -20,20 +20,14 @@
   "bugs": "https://github.com/nosir/cleave.js/issues",
   "homepage": "https://github.com/nosir/cleave.js",
   "docs": "https://github.com/nosir/cleave.js/blob/master/README.md",
-  "main": "src/Cleave.js",
+  "main": "dist/cleave.js",
```
Comment (nosir) [inline package.json]: Can we not change this one?
The vanilla JS component doesn't need any ES6 stuff, so won't have problem when it is built.

Comment (chmanie) [inline package.json]: I did this on purpose but I don't remember anymore why (didn't work for me). I'll come back to you on this, soon!

**Comment by nosir** [comment]
@chmanie I think this is fine, will merge this.
Ideally setup browserify-shim in package itself can reduce the finial size a lot if we bundling by using browserify-shim out of the box. But well, I guess it makes the setup too complicated.

**Pull request merged**